packages/flutter/lib/src/widgets/selection_gestures.dart

```diff
@@ -0,0 +1,1251 @@
+///  * [TapDragDownDetails], the details for [GestureTapDragDownCallback].
+///  * [TapDragUpDetails], the details for [GestureTapDragUpCallback].
+///  * [TapDragUpdateDetails], the details for [GestureTapDragUpdateCallback].
+///  * [TapDragEndDetails], the details for [GestureTapDragEndCallback].
+class TapDragStartDetails {
+  /// Creates details for a [GestureTapDragStartCallback].
+  ///
+  /// The [globalPosition] argument must not be null.
+  TapDragStartDetails({
+    this.sourceTimeStamp,
+    this.globalPosition = Offset.zero,
+    Offset? localPosition,
+    this.kind,
+    required this.consecutiveTapCount,
+    required this.keysPressedOnDown,
+  }) : assert(globalPosition != null),
+       localPosition = localPosition ?? globalPosition;
+
+  /// Recorded timestamp of the source pointer event that triggered the drag
+  /// event.
+  ///
+  /// Could be null if triggered from proxied events such as accessibility.
+  final Duration? sourceTimeStamp;
+
+  /// The global position at which the pointer contacted the screen.
+  ///
+  /// Defaults to the origin if not specified in the constructor.
+  ///
+  /// See also:
+  ///
+  ///  * [localPosition], which is the [globalPosition] transformed to the
+  ///    coordinate space of the event receiver.
+  final Offset globalPosition;
+
+  /// The local position in the coordinate system of the event receiver at
+  /// which the pointer contacted the screen.
+  ///
+  /// Defaults to [globalPosition] if not specified in the constructor.
+  final Offset localPosition;
+
+  /// The kind of the device that initiated the event.
+  final PointerDeviceKind? kind;
+
+  /// If this tap is in a series of taps, then this value represents
+  /// the number in the series this tap is.
+  final int consecutiveTapCount;
+
+  /// The keys that were pressed when the most recent [PointerDownEvent] occurred.
+  final Set<LogicalKeyboardKey> keysPressedOnDown;
+
+  @override
+  String toString() => '${objectRuntimeType(this, 'TapDragStartDetails')}($globalPosition)';
+}
+
+/// {@macro flutter.gestures.dragdetails.GestureDragUpdateCallback}
+///
+/// The consecutive tap count when the drag was initiated is given by [TapDragUpdateDetails.consecutiveTapCount].
+///
+/// Used by [TapAndDragGestureRecognizer.onUpdate].
+typedef GestureTapDragUpdateCallback = void Function(TapDragUpdateDetails details);
+
+/// Details for [GestureTapDragUpdateCallback], such as the tap count within
+/// the series.
+///
+/// See also:
+///
+///  * [TapAndDragGestureRecognizer], which passes this information to its
+///    [TapAndDragGestureRecognizer.onUpdate] callback.
+///  * [TapDragDownDetails], the details for [GestureTapDragDownCallback].
+///  * [TapDragUpDetails], the details for [GestureTapDragUpCallback].
+///  * [TapDragStartDetails], the details for [GestureTapDragStartCallback].
+///  * [TapDragEndDetails], the details for [GestureTapDragEndCallback].
+class TapDragUpdateDetails {
```

Comment:
I thought about it, I actually tried extending them initially but wasn't super happy how it looked. I'm not sure in what direction to go with that but am open to suggestions. Currently none of the details objects extend each other so I followed that pattern. I think it might be a nice-to-have to refactor them to use a common base class, but also wasn't sure if I wanted to add more to an already large PR.

Reply:
That sounds good to me.  I'll keep it in mind as we move forward with this in case any idea strikes me, but it definitely makes sense as-is.